cmd/brew-postgresql-upgrade-database.rb

```diff
@@ -57,12 +57,30 @@
     server_stopped = true
   end
 
+  # get 'lc_collate' from old DB"
+  unless quiet_system "#{old_bin}/pg_ctl", "-w", "-D", datadir, "status"
+    system "#{old_bin}/pg_ctl", "-w", "-D", datadir, "start"
+  end
+
+  sql_for_lc_collate = "SELECT setting FROM pg_settings WHERE name LIKE 'lc_collate';"
+  sql_for_lc_ctype = "SELECT setting FROM pg_settings WHERE name LIKE 'lc_ctype';"
+  lc_collate = Utils.popen_read("#{old_bin}/psql", "postgres", "-qtAc", sql_for_lc_collate).strip
+  lc_ctype = Utils.popen_read("#{old_bin}/psql", "postgres", "-qtAc", sql_for_lc_ctype).strip
+
+  if quiet_system "#{old_bin}/pg_ctl", "-w", "-D", datadir, "status"
+    system "#{old_bin}/pg_ctl", "-w", "-D", datadir, "stop"
+  end
+
   ohai "Moving #{name} data from #{datadir} to #{old_datadir}..."
   FileUtils.mv datadir, old_datadir
   moved_data = true
 
   (var/"postgres").mkpath
-  system "#{bin}/initdb", "#{var}/postgres"
+  if lc_collate.empty? || lc_ctype.empty?
```

Comment:
Is there a case where one could be empty and the other not? May want to populate an `args` array and add to it depending on what's present (and they could share a system "#{bin}/initdb", *args, "#{var}/postgres"` call)

Reply:
I thought that case is quite rare and I didn't want to spend multiple lines for it.

Thanks for the advice, I didn't know the empty array doesn't harm `system` call. I changed the commit.